dot/network/service.go

```diff
@@ -33,18 +35,26 @@ var _ services.Service = &Service{}
 
 // Service describes a network service
 type Service struct {
-	ctx               context.Context
-	cfg               *Config
-	host              *host
-	mdns              *mdns
-	status            *status
-	gossip            *gossip
-	msgRec            <-chan Message
-	msgSend           chan<- Message
-	noBootstrap       bool
-	noMDNS            bool
-	noStatus          bool            // internal option
-	noGossip          bool            // internal option
+	ctx    context.Context
+	cfg    *Config
+	host   *host
+	mdns   *mdns
+	status *status
+	gossip *gossip
+
+	// Channels for inter-process communication
+	// as well as a lock for safe channel closures
+	msgRec  <-chan Message
+	msgSend chan<- Message
+	lock    sync.Mutex
+	closed  bool
```

Comment:
I wonder if we can clean up the properties of each service using safe message send and receive (`core` and `network`) packagees within a separate lib submodule that both each service uses rather than the safe send and receive being a method appended to each service...